src/documents/tasks.py

```diff
@@ -292,3 +296,25 @@ def update_document_archive_file(document_id):
         )
     finally:
         parser.cleanup()
+
+
+@shared_task
+def empty_trash(doc_ids=None):
+    documents = (
+        Document.deleted_objects.filter(id__in=doc_ids)
+        if doc_ids is not None
+        else Document.deleted_objects.filter(
+            deleted_at__lt=timezone.localtime(timezone.now())
+            - timedelta(
+                days=settings.EMPTY_TRASH_DELAY,
+            ),
+        )
+    )
+
+    # Temporarily connect the cleanup handler (hard_delete calls delete)
+    models.signals.post_delete.connect(cleanup_document_deletion, sender=Document)
```

Comment:
Just in case, lets wrap this in try/finally so the signal is for sure re-connected